manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -153,23 +178,53 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 			}
 
 			if len(tr.dirty)+len(tr.cleanup) > maxDirty {
+				// stop the timer, so we don't fire it. if we get another event
+				// after we do this cleaning, we will reset the timer then
 				timer.Stop()
+				// if the timer had fired, drain out the value.
+				select {
```

Comment:
I don't think it's worthwhile to move this to a function. it's essentially just an if statement for a channel read. sort of like

```
if HasValue(channel) {
    <-channel
}
```

but you can't do such an if statement, so you do this select with an empty default.

Reply:
alternatively you can do

```go
if !timer.Stop() {
    <-timer.C
}
```

which is how the docs drain a channel, but for some reason doing it that way was deadlocking, so i did it the easy way instead of trying to figure out what was weird.